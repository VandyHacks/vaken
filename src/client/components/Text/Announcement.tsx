import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import FloatingPopup from '../Containers/FloatingPopup';
import STRINGS from '../../assets/strings.json';
import Notification from '../../assets/img/notification.svg';

const Msg = styled.p`
	font-size: 1rem;
	color: #ffffff;
	white-space: pre-line;
	grid-area: text;
	padding-left: 1.5rem;
`;

const AnnouncementImg = styled.img`
	width: 7rem;
	grid-area: image;
`;

const Layout = styled.div`
	display: grid;
	grid-template-columns: 7rem auto;
	grid-template-rows: auto;
	grid-template-areas: 'image text';
`;

export interface Props {
	value: string;
}

export const Announcement: FunctionComponent<Props> = (props: Props): JSX.Element => {
	const { value } = props;

	return (
		<FloatingPopup
			borderRadius="1rem"
			width="35rem"
			backgroundOpacity="1"
			backgroundColor={STRINGS.ACCENT_COLOR}
			marginBottom="1rem"
			padding="1.5rem">
			<Layout>
				<AnnouncementImg src={Notification} />
				<Msg>{value}</Msg>
			</Layout>
		</FloatingPopup>
	);
};

export default Announcement;
