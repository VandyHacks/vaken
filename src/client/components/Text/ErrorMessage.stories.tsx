import React from 'react';
import { Meta, Story } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import {
	ErrorMessage as Component,
	ErrorMessageProps as Props,
	GraphQLErrorMessage as GQLComponent,
	GraphQLErrorMessageProps as GQLProps,
} from './ErrorMessage';
import { theme } from '../../app';

export default {
	title: 'Components/Text/Error Message',
	component: Component,
} as Meta;

export const ErrorMessage: Story<Props> = args => <Component {...args} />;
ErrorMessage.args = { children: <p>This is an error message</p> };

export const GraphQLErrorMessage: Story<GQLProps> = args => (
	<ThemeProvider theme={theme}>
		<MemoryRouter>
			<GQLComponent {...args} />
		</MemoryRouter>
	</ThemeProvider>
);
GraphQLErrorMessage.args = { text: 'This is an error message' };

export const GraphQLErrorMessageWithLongText: Story<GQLProps> = args => (
	<GraphQLErrorMessage {...args} />
);
GraphQLErrorMessageWithLongText.args = {
	text:
		"This is an error message with really really long text that just keeps going. You think that error messages could be concise and simple to understand because what happened? An Error. No ordinary user will be able to fix an error message, and a real dev would just open the dev tools, but no. Long messages are a must. What happens if it is autogenerated by Apollo? Need to have support for it. Object string dump? Gotta support it. Our entire database of PII? Let's at least make sure it's possible.",
};
