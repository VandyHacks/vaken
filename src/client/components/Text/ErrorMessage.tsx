import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import STRINGS from '../../assets/strings.json';
import SadFace from '../../assets/img/sad_face.svg';
import { Button } from '../Buttons/Button';

const Rectangle = styled.div`
	border: 0.1rem solid ${STRINGS.WARNING_COLOR};
	border-radius: 8px;
	display: inline-block;
	text-align: center;
	color: ${STRINGS.WARNING_COLOR};
	font-family: 'Roboto', sans-serif;
	font-weight: 500;
	font-size: 1.5rem;
	display: inline-block;
	padding: 1rem;
`;

const StyledP = styled.p`
	white-space: pre-line;
`;

export interface ErrorMessageProps {
	children?: JSX.Element;
}

export const ErrorMessage: FunctionComponent<ErrorMessageProps> = (
	props: ErrorMessageProps
): JSX.Element => {
	const { children } = props;
	return (
		<Rectangle>
			<img src={SadFace} alt="Sad face" />
			{children}
		</Rectangle>
	);
};

export interface GraphQLErrorMessageProps {
	text?: string;
}

export const GraphQLErrorMessage: FunctionComponent<GraphQLErrorMessageProps> = (
	props: GraphQLErrorMessageProps
): JSX.Element => {
	const { text } = props;
	return (
		<ErrorMessage>
			<>
				<StyledP>{text}</StyledP>
				<Button warning long large linkTo="/dashboard">
					Return to Dashboard
				</Button>
			</>
		</ErrorMessage>
	);
};
