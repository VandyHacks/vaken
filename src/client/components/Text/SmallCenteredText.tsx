import styled from 'styled-components';

export interface Props {
	/** Text color */
	color?: string;
	fontFamily?: string;
	fontSize?: string;
	fontWeight?: string;
	/** Margin applied to the top and bottom */
	margin?: string;
	/** The text to display inside the tag */
	children: React.ReactChild;
}

export const SmallCenteredText = styled.h1`
	font-family: ${({ fontFamily = "'Roboto', sans-serif" }: Props): string => fontFamily};
	font-weight: ${({ fontWeight = '300' }: Props): string => fontWeight};
	font-size: ${({ fontSize = '1.5rem' }: Props): string => fontSize};
	color: ${({ color = 'white' }: Props): string => color};
	margin: ${({ margin = '0' }: Props): string => margin} 0;
`;

export default SmallCenteredText;
