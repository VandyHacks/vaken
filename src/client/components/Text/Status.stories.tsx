import React from 'react';
import { Meta, Story } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { ComponentProps as Props, Status as Component } from './Status';
import { ApplicationStatus } from '../../generated/graphql';

export default {
	title: 'Components/Text/Status',
	component: Component,
} as Meta;

export const Status: Story<Props> = args => <Component {...args} />;
Status.args = { generateColor: () => 'rebeccapurple', value: ApplicationStatus.Accepted };
