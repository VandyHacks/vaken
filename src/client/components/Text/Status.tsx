import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import { ApplicationStatus } from '../../generated/graphql';
import STRINGS from '../../assets/strings.json';

interface Props {
	borderRadius?: string;
	fontColor?: string;
	fontFamily?: string;
	fontSize?: string;
	fontWeight?: number;
	textAlign?: string;
	width?: string;
}

export interface ComponentProps extends Props {
	/** Function which maps `ApplicationStatus`s to CSS color strings */
	generateColor?: (value: ApplicationStatus) => string;
	value: ApplicationStatus;
}

interface StyledProps extends Props {
	backgroundColor?: string;
}

const StyledDiv = styled('div')`
	font-family: ${(props: StyledProps) => props.fontFamily || "'Roboto Condensed'"};
	font-size: ${(props: StyledProps) => props.fontSize || '1rem'};
	font-weight: ${(props: StyledProps) => props.fontWeight || 500};
	color: ${(props: StyledProps) => props.fontColor || 'white'};
	background-color: ${(props: StyledProps) => props.backgroundColor || STRINGS.ACCENT_COLOR};
	text-align: ${(props: StyledProps) => props.textAlign || 'center'};
	border-radius: ${(props: StyledProps) => props.borderRadius || '8px'};
	width: ${(props: StyledProps) => props.width || '5rem'};
	text-transform: lowercase;

	&::first-letter {
		text-transform: capitalize;
	}
`;

export const Status: FunctionComponent<ComponentProps> = ({
	value,
	generateColor,
	...props
}: ComponentProps): JSX.Element => (
	<StyledDiv {...props} backgroundColor={generateColor && generateColor(value)}>
		{value}
	</StyledDiv>
);

export default Status;
