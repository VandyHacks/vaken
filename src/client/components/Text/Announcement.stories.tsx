import React from 'react';
import { Meta, Story } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { Announcement as Component, Props } from './Announcement';

export default {
	title: 'Components/Text/Announcement',
	component: Component,
	args: {
		value: 'This is an announcement',
	},
} as Meta;

export const Announcement: Story<Props> = args => <Component {...args} />;
