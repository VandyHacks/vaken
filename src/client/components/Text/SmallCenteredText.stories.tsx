import React from 'react';
import { Meta, Story } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { SmallCenteredText as Component, Props } from './SmallCenteredText';

export default {
	title: 'Components/Text/Small Centered Text',
	component: Component,
	parameters: { backgrounds: { default: 'dark' } },
} as Meta;

export const SmallCenteredText: Story<Props> = args => <Component {...args} />;
SmallCenteredText.args = { children: 'This is small, centered text.' };
