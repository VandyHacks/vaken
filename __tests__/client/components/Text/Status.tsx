import React from 'react';
import renderer from 'react-test-renderer';

import { Status } from '../../../../src/client/components/Text/Status';

describe('Test Status', () => {
	it('renders correctly', async () => {
		const component = renderer
			.create(<Status value="mock" generateColor={() => 'mock'} />)
			.toJSON();
		expect(component).toMatchSnapshot();
	});
});
