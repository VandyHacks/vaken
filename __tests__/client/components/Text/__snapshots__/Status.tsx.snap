// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Status renders correctly 1`] = `
<div
  className="sc-bdVaJa fhDPKM"
  value="mock"
>
  mock
</div>
`;
